### PR TITLE
WH-89: Validation and Sanitization

### PR DESCRIPTION
**Pull request by abp6318**
#### Jira
[https://watchhill.atlassian.net/browse/WH-89](https://watchhill.atlassian.net/browse/WH-89)

#### TODO
- [x] Validation for passwords
- [x] Validation for usernames
- [x] Sanitization for removing `$ ` (similar to [this article](https://www.npmjs.com/package/mongo-sanitize?activeTab=readme))
- [x] Update all POST requests
- [x] Update all PUT requests (might be tricky / I need help)
- [x] Add and update tests (consult Vicky possibly)
- [x] Validate 4 types of IDs
- [x] Validate DateTime strings
- [x] Document
- [x] Refactor

Please test using Postman and not the frontend. Fixing the frontend will be a separate PR with more quality of life changes.

### CODE REVIEW

File: routes/game/gameRoutes.js [hunk 1]
```diff
@@ -39,7 +43,7 @@ router.put('/', async (req, res) => {
         const updGame = await GameInfo.findOne({_id: ObjectId(req.body._id)});
 
         if (updGame) {
-            const {updatedData} = req.body;
+            const { updatedData } = req.body; // TODO: Validation for this is a mess
```
Comment (abp6318) [inline routes/game/gameRoutes.js]: @aidan-lemay @vsoler27 Any ideas on how to validate input for a JSON object with dynamic properties?

Comment (vsoler27) [reply routes/game/gameRoutes.js]: other than checking for properties and validating in ifs, not really. i need to think abt it a bit

**Review by vsoler27 — COMMENTED** [review]
other than the PUT stuff this looks good

**Review by vsoler27 — APPROVED** [review]
ur honor shes a beaut

**Pull request merged**